nba-front-end/src/components/playerDataGrid/playerTableLoader.tsx

```diff
@@ -19,24 +20,42 @@ function FilledPlayerTable() {
   // we call this as <TableLoading isLoading={appState.loading} playerList={appState.playerList} />
   // isLoading={appState.loading} says that the boolean for whether data is available is 'appState.loading' variable
   // playerList={appState.playerList} says that (if the data is available) send the appState.playerList to the DataGridPlayers Component
-  const TableLoading = WithTableLoading(DataGridPlayers);
+  const TableLoading = (DataGridPlayers);
   const [appState, setAppState] = useState<PlayerProps>({
-    loading: false,
     playerList: [],
   });
+  // defines a state for whenever an error occurs
+  const [errorMessage, setErrorMessage] = useState("");
+    // defines a state for when the api is fetching data for players
+  const [isLoading, setLoading] = useState(false);
 
   // this is the call to the API to get the player data
   useEffect(() => {
-    setAppState({ loading: true, playerList: [] });
-    api.get('players/get-all').subscribe((resp) => {
-        setAppState({ loading: false, playerList: resp as Player[] });
-      });
-  }, [setAppState]);
+    setLoading(true);
+    setErrorMessage("");
+    setAppState({ playerList: [] });
+    api.get('players/get-all').toPromise().then((resp) => {
+        setLoading(false);
+        setAppState({ playerList: resp as Player[] });
+        })
+  // this catches any errors that may occur while fetching for player data
+        .catch(error => { console.log(error) 
+        setLoading(false);
+  // this sets 'errorMessage' into the error that has occured
+        setErrorMessage(error);
+        })
+      }, [setAppState]);
 
   return (
+    <React.Fragment>
```

Comment:
this <React.Fragment></React.Fragment>
Can be changed to <></> for short hand I think